File: src/services/schema-export/tests/models/all/kotlin/IndexedTypes.kt
// Please note : @LinkingObjects and default values are not represented in the schema and thus will not be part of the generated models
package your.package.name.here

import io.realm.RealmObject
import io.realm.annotations.Index
import java.util.Date
import io.realm.annotations.PrimaryKey

open class IndexedTypes : RealmObject() {

    @Index
    var boolIndexed: Boolean = false
    @PrimaryKey
    var intIndexed: Long = 0
    @Index
    var stringIndexed: String = ""
    @Index
    var dateIndexed: Date = Date()

}

File: src/services/schema-export/tests/models/all/cs/IndexedTypes.cs
// Please note : [Backlink] properties and default values are not represented
// in the schema and thus will not be part of the generated models

using System;
using System.Collections.Generic;
using Realms;

namespace MyProject.Models
{
    public class IndexedTypes : RealmObject
    {
        [Indexed]
        [MapTo("boolIndexed")]
        public bool BoolIndexed { get; set; }

        [PrimaryKey]
        [MapTo("intIndexed")]
        public long IntIndexed { get; set; }

        [Indexed]
        [Required]
        [MapTo("stringIndexed")]
        public string StringIndexed { get; set; }

        [Indexed]
        [MapTo("dateIndexed")]
        public DateTimeOffset DateIndexed { get; set; }
    }
}
